common/src/main/java/org/apache/comet/vector/CometVector.java

```diff
@@ -125,14 +144,14 @@ byte[] getBinaryDecimal(int i) {
 
   /** Reads a 16-byte byte array which are encoded big-endian for decimal128. */
   public byte[] copyBinaryDecimal(int i, byte[] dest) {
-    long valueBufferAddress = getValueVector().getDataBuffer().memoryAddress();
-    Platform.copyMemory(
-        null,
-        valueBufferAddress + (long) i * DECIMAL_BYTE_WIDTH,
-        dest,
-        Platform.BYTE_ARRAY_OFFSET,
-        DECIMAL_BYTE_WIDTH);
+    if (DECIMAL_BYTES_ALL == null) {
+      ValueVector vector = getValueVector();
+      DECIMAL_BYTES_ALL = new byte[vector.getBufferSize()];
+      copyBuffer(vector, DECIMAL_BYTES_ALL);
+    }
     // Decimal is stored little-endian in Arrow, so we need to reverse the bytes here
+    System.arraycopy(
+        DECIMAL_BYTES_ALL, i * DECIMAL_BYTE_WIDTH, DECIMAL_BYTES, 0, DECIMAL_BYTE_WIDTH);
```

Comment:
Should this be copying to `dest` rather than `DECIMAL_BYTES`? 

```suggestion
    System.arraycopy(
        DECIMAL_BYTES_ALL, i * DECIMAL_BYTE_WIDTH, data, 0, DECIMAL_BYTE_WIDTH);
```